pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/TimeSchema.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.impl.schema;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.SchemaSerializationException;
+import org.apache.pulsar.common.schema.SchemaInfo;
+import org.apache.pulsar.common.schema.SchemaType;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.sql.Time;
+
+/**
+ * A schema for `java.sql.Time`.
+ */
+public class TimeSchema implements Schema<Time> {
+   public static TimeSchema of() {
+      return INSTANCE;
+   }
+
+   private static final TimeSchema INSTANCE = new TimeSchema();
+   private static final SchemaInfo SCHEMA_INFO = new SchemaInfo()
+         .setName("Time")
+         .setType(SchemaType.TIME)
+         .setSchema(new byte[0]);
+
+   @Override
+   public byte[] encode(Time message) {
+      if (null == message) {
+         return null;
+      }
+
+      try(ByteArrayOutputStream bos = new ByteArrayOutputStream(); ObjectOutputStream objOut = new ObjectOutputStream(bos)) {
```

Comment:
I am not sure this is the right approach for serializing the date. I think a better approach is to convert the data to timestamp (in millis), so that you can just serialize the long number. You can have a property in the schema about the date format, so that you can use that for converting it from and to the timestamp.

Reply:
How to modify this?
```java
@Override
   public byte[] encode(Time message) {
      if (null == message) {
         return null;
      }

      Long time = message.getTime();
      return LongSchema.of().encode(time);
   }

   @Override
   public Time decode(byte[] bytes) {
      if (null == bytes) {
         return null;
      }

      Long decode = LongSchema.of().decode(bytes);
      return new Time(decode);
   }
```

Reply:
yes. that's what I had in my mind.

Reply:
ok, thanks